File: backend/src/main/java/turip/country/controller/dto/CountryResponse.java
package turip.country.controller.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import turip.country.domain.Country;

public record CountryResponse(
        Long id,
        @JsonProperty("countryName")
        String name,
        @JsonProperty("countryImageUrl")
        String imageUrl
) {

    public static CountryResponse from(Country country) {
        return new CountryResponse(
                country.getId(),
                country.getName(),
                country.getImageUrl()
        );
    }
}

File: mobile/core/data/src/commonMain/kotlin/com/on/turip/core/data/dto/region/CountryResponse.kt
package com.on.turip.core.data.dto.region

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CountryResponse(
    @SerialName("countryImageUrl")
    val countryImageUrl: String,
    @SerialName("countryName")
    val countryName: String,
    @SerialName("id")
    val id: Int,
)
